### ISSUE TITLE
Default password not working, environment vars not working either

### ISSUE THREAD

**Issue by fabstao** [bitnami/bitnami-docker-jenkins#51]
**Description**

Hi!

Can´t login to Jenkins image using docker-compose

**Steps to reproduce the issue:**

1. Download docker-compose.yaml for Bitnami Jenkins and launch with docker-compose :
```
$ curl -sSL https://raw.githubusercontent.com/bitnami/bitnami-docker-jenkins/master/docker-compose.yml > docker-compose.yml
$ docker-compose up -d
```
Edit docker-compose and map ports 80:8080 , 443:8443 and 50000:50000

I can't use port 80

2. Try to login to: http://localhost:80
user: user
password: bitnami
Error message:
Invalid username or password

3. Destroy environment and setup again:
```
$ docker-compose down
$ docker rmi <jenkins image id>
```
Edit docker-compose.yaml and add:
```
    environment:
      - JENKINS_USER="saitama"
      - JENKINS_PASSWORD="OnePunchMan"
```
Then
Launch with `docker-compose up -d` again

**Describe the results you received:**

user: saitama
password: OnePunchMan
Error message:
Invalid username or password

**Describe the results you expected:**

Login successful

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version**

- Output of `docker version`:
```
$ docker --version
Docker version 18.09.6, build 481bc77156
$ 
$ docker-compose --version
docker-compose version 1.24.0, build 0aa59064
$ 
```

- Output of `docker info`:

```
$ docker info
Containers: 2
 Running: 2
 Paused: 0
 Stopped: 0
Images: 2
Server Version: 18.09.6
Storage Driver: overlay2
 Backing Filesystem: xfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: bb71b10fd8f58240ca47fbb579b9d1028eea7c84
runc version: 2b18fe1d885ee5083ef9f0838fee39b62d653e30
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-957.21.3.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 3
Total Memory: 5.337GiB
Name: centos7base.local
ID: O35A:UEFK:SNH7:BHTF:SIK7:3QZW:VJSX:4RSC:TJHY:PLML:SPFR:BZZN
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
HTTP Proxy: http://proxy:3128/  HTTPS_PROXY=http://proxy:3128
NO_PROXY=10.0.0.0/8,localhost,192.168.0.0/16
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

**Comment by tompizmor** [comment]
Hi @fabstao,

So you have reported two issues:

**1. Unable to login using default credentials**

You are using `docker-compose down` to clean the environment and start again but that does not clean the volume so you are reusing the volumes (with persisted username and password) from the first initialization. 

Use `docker-compose down -v` and try again with the default credentials (user/bitnami)

**2. Environment variables to set username and password not working**

You are using a wrong environment variable for the username. It should be `JENKINS_USERNAME`.
Finally, you are setting your username and password with double-quotes. Unless you want them as [art of the username or password, you should delete them.

The following docker-compose file should work fine for you.

```
version: '2'
services:
  jenkins:
    image: 'bitnami/jenkins:2'
    environment:
      - JENKINS_USERNAME=saitama
      - JENKINS_PASSWORD=OnePunchMan
    ports:
      - '80:8080'
      - '443:8443'
      - '50000:50000'
    volumes:
      - 'jenkins_data:/bitnami'
volumes:
  jenkins_data:
    driver: local
```

**Comment by fabstao** [comment]
Hi! Thanks let me try again and get you back!

**Comment by dkirrane** [comment]
I've tried this and it doesn't work

**Comment by x4x** [comment]
i have exactly the same issue.

environment variables `JENKINS_USER` and `JENKINS_PASSWORD` have no efect.
and im unable to login using default credentials since i switched from version 2 to latest.

**Comment by tompizmor** [comment]
Hi @x4x,

The environment variable to set the username is  `JENKINS_USERNAME`, not `JENKINS_USER`. Can you try that?

Also, please make sure you clean the volumes with `docker-compose down -v` or the login credentials will be stored from the first deployment.

**Comment by x4x** [comment]
hi @tompizmor,

sorry, I copied it wron i used `JENKINS_USERNAME` in my config.

after some tesing i found out that i cant login if i use `JENKINS_USERNAME=admin` every other name works fine.

**Comment by tompizmor** [comment]
I see! Thanks for updating the issue.

Probably Jenkins doesn't allow using some well-known administrator users for security reasons.

**Comment by dexdinh91** [comment]
> hi @tompizmor,
> 
> sorry, I copied it wron i used `JENKINS_USERNAME` in my config.
> 
> after some tesing i found out that i cant login if i use `JENKINS_USERNAME=admin` every other name works fine.

I have the same issue with `JENKINS_USERNAME=admin`, thanks!

**Comment by joancafom** [comment]
Hey @dexdinh91,

Were you able to solve the issue?

Thanks!

**Comment by GraysonCAdams** [comment]
Hello, I have this issue as well.

Using a persistent volume, changing the JENKINS_USERNAME or JENKINS_PASSWORD environment variables has no bearing on the admin login after Jenkins was initially built.

**Comment by GraysonCAdams** [comment]
Is this intended behavior?

**Comment by juan131** [comment]
> Is this intended behavior?

Correct. When you persist the data and it's detected when booting the container, the initialisation logic that configures Jenkins will be skipped. In other words, these environment variables are only considered during the 1st boot of the container.
